pageserver/src/tenant/layer_map/historic_layer_coverage.rs

```diff
@@ -391,6 +398,16 @@ pub struct BufferedHistoricLayerCoverage<Value> {
 
     /// All current layers. This is not used for search. Only to make rebuilds easier.
     layers: BTreeMap<LayerKey, Value>,
+
+    /// Redundant layers are ones that are completely covered by a union of other layers.
+    /// If two layers are identical only one of them will be marked as redundant, such that
+    /// it is always safe to remove all redundant layers without seeing any difference in
+    /// results.
```

Comment:
If there are two sets of layers that cover the same key-lsn space, which set is marked as redundant? For example, right after compaction has created all the L1 layers, but hasn't removed any of the L0 layers yet.

The PR comment says:

> I change the insertion order of the layer map so that when pageserver wakes up and finds both the L1 and the constituent L0s (which should have been deleted but haven't been), the L0s are marked redundant, not the L1.

But that's not really made clear anywhere in the code.